internal/services/mysql/mysql_flexible_server_resource.go

```diff
@@ -752,6 +756,14 @@ func expandArmServerNetwork(d *pluginsdk.ResourceData) *servers.Network {
 		network.PrivateDnsZoneResourceId = utils.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("public_network_access_enabled"); ok {
```

Comment:
Because this is defaulted, we shouldn't have to check if it exists and we can just set the value based on `d.Get("public_network_access_enabled")`